### PR TITLE
RBAC

### PR DESCRIPTION
**Pull request by brancz**
I started writing the RBAC guide and realized we need to be able to pass the `ServiceAccountName` that ends up in the `PodTemplate`. So I added it and finished the RBAC guide and example manifests.

@fabxc 

/cc @ericchiang

### CODE REVIEW

**Comment by fabxc** [comment]
Looks good AFAICS.

**Comment by brancz** [comment]
I'll let @ericchiang give this a read before I merge this.

File: Documentation/rbac.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# RBAC
+
+RBAC for the Prometheus Operator involves two parts, RBAC rules for the
+Operator itself and RBAC rules created by the Operator as Prometheus itself
```
Comment (fabxc) [inline Documentation/rbac.md]: This was on an old revision and I didn't hit "submit" yesterday. Fixed already.

File: Documentation/rbac.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- apiGroups: [""]
+  resources:
+  - nodes
+  - services
+  - endpoints
+  - pods
+  verbs: ["get", "list", "watch"]
+```
+
+> Note: A cluster admin is required to create this `ClusterRole` and create a
```
Comment (ericchiang) [inline Documentation/rbac.md]: Assuming that the prometheus operator always creates prometheuses with the "default" service account. Maybe give an example of a `ClusterRoleBinding`?

```
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1alpha1
metadata:
  name: prometheus-( target namespace ) # Name isn't required to be of this form.
subjects:
  - kind: ServiceAccount
    name: default
    namespace: ( target namespace ) # Namespace that prometheus is deployed in.
roleRef:
  kind: ClusterRole
  name: prometheus
  apiGroup: rbac.authorization.k8s.io
```

Also, maybe it'd be good to create a `prometheus` service account?

Comment (ericchiang) [inline Documentation/rbac.md]: Ah I see bellow that this is configurable.

**Comment by ericchiang** [comment]
one comment, other than that lgtm

**Comment by brancz** [comment]
Usage example added.

**Pull request merged**